frontend-react/src/contexts/Toast/index.tsx

```diff
@@ -0,0 +1,87 @@
+import { toast } from "react-toastify";
+import React, { createContext, useCallback, useContext, useMemo } from "react";
+
+import { useSessionContext } from "../Session";
+
+export interface ToastCtx {
+    toast: (
+        msgOrErr: Error | React.ReactNode,
+        type: "success" | "warning" | "error" | "info",
+    ) => void;
+}
+
+export const ToastContext = createContext<ToastCtx>({ toast: () => void 0 });
+
+export const showToast = (
+    message: React.ReactNode,
+    type: "success" | "warning" | "error" | "info",
+) => {
+    const toastId = `id_${message}`.replace(/\W/gi, "_").substring(0, 512);
+    const msg = message instanceof Error ? message.message : message;
+
+    // basically the USWDS Alert UI
+    toast(
+        <div
+            className={`usa-alert usa-alert--slim usa-alert--no-icon usa-alert--${type} rs-alert-toast`}
+            data-testid="alerttoast"
+        >
+            <div className="usa-alert__body">
+                {msg && <h2 className="usa-alert__heading">{msg}</h2>}
+            </div>
+        </div>,
+        {
+            toastId,
+            autoClose: 5000,
+            delay: 10,
+            closeButton: false,
+            position: "bottom-center",
+            hideProgressBar: true,
+        },
+    );
+    toast.clearWaitingQueue(); // don't pile up messages
+};
+
+export const useToast = () => useContext(ToastContext);
+
+export function ToastProvider({ children }: React.PropsWithChildren) {
```

Comment:
Style preference: we should use Direct Imports for all of our methods since it's much less verbose, and we can easily see at the top of the file what React imports we're using so: 

```
import React, { PropsWithChildren } from 'react';
```

Reply:
ie, we use this syntax when using `useEffect`, `useState` etc.